spec/forms/application_form_spec.rb

```diff
@@ -54,4 +54,110 @@
       end
     end
   end
+
+  describe ".date_attribute" do
+    subject(:example_form) { ExampleForm.new }
+
+    before do
+      example_form_class = Class.new(ApplicationForm) do
+        date_attribute :date_of_birth
+      end
+
+      stub_const("ExampleForm", example_form_class)
+    end
+
+    it "adds getters and setters for the attribute suffixed with (1i), (2i), and (3i)" do
+      expect(example_form).to respond_to("date_of_birth(1i)")
+      expect(example_form).to respond_to("date_of_birth(2i)")
+      expect(example_form).to respond_to("date_of_birth(3i)")
+
+      expect(example_form).to respond_to("date_of_birth(1i)=")
+      expect(example_form).to respond_to("date_of_birth(2i)=")
+      expect(example_form).to respond_to("date_of_birth(3i)=")
+
+      example_form.send("date_of_birth(1i)=", 2024)
+      example_form.send("date_of_birth(2i)=", 5)
+      example_form.send("date_of_birth(3i)=", 2)
+
+      expect(example_form.send("date_of_birth(1i)")).to eq(2024)
+      expect(example_form.send("date_of_birth(2i)")).to eq(5)
+      expect(example_form.send("date_of_birth(3i)")).to eq(2)
```

Comment:
It would be worth using `public_send` to demonstrate that these are public methods. I think it's generally better to use `public_send` whenever possible, unless you really need to use `send`.

```suggestion
      example_form.public_send("date_of_birth(1i)=", 2024)
      example_form.public_send("date_of_birth(2i)=", 5)
      example_form.public_send("date_of_birth(3i)=", 2)

      expect(example_form.public_send("date_of_birth(1i)")).to eq(2024)
      expect(example_form.public_send("date_of_birth(2i)")).to eq(5)
      expect(example_form.public_send("date_of_birth(3i)")).to eq(2)
```

The same applies to the other examples in this spec.